### PR TITLE
Drop KonnectivityTunnel feature gate

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area networking open-source
/kind cleanup

**What this PR does / why we need it**:
With #3771 and #3812, the Gardener community has agreed to pursue another VPN solution for the future and to drop `KonnectivityTunnel`. This PR cleans up the feature and all related code of it.

**Which issue(s) this PR fixes**:
Fixes #

**Release note**:

```breaking operator
The `KonnectivityTunnel` feature gate in gardenlet has been dropped and removed from the code. If you upgrade to this Gardener version make sure that the feature gate is disabled and that all shoots were reconciled after you disabled it.
```

### CODE REVIEW

File: charts/images.yaml [hunk 1]
```diff
@@ -90,10 +90,6 @@ images:
   sourceRepository: github.com/gardener/vpn2
   repository: eu.gcr.io/gardener-project/gardener/vpn-seed-server
   tag: "0.4.0"
-- name: konnectivity-server
-  sourceRepository: github.com/gardener/replica-reloader
```
Comment (vpnachev) [inline charts/images.yaml:94]: I have opened https://github.com/gardener/replica-reloader/issues/7 to follow up after this PR is merged.

**Review by vpnachev — DISMISSED** [review]
/lgtm

**Comment by vpnachev** [comment]
I guess it will be nice to add a short note in https://github.com/gardener/gardener/blob/master/docs/proposals/11-apiserver-network-proxy.md that it was deprecated and removed in favor of https://github.com/gardener/gardener/blob/master/docs/proposals/14-reversed-cluster-vpn.md. WDYT?

**Comment by rfranzke** [comment]
> I guess it will be nice to add a short note in https://github.com/gardener/gardener/blob/master/docs/proposals/11-apiserver-network-proxy.md that it was deprecated and removed in favor of https://github.com/gardener/gardener/blob/master/docs/proposals/14-reversed-cluster-vpn.md. WDYT?

Not sure, those are proposal documents and we usually do not keep them up-to-date (e.g., GEP-1 was not touched since initial contribution).

**Comment by vpnachev** [comment]
I see `status` field in the GEP template https://github.com/gardener/gardener/blob/master/docs/proposals/00-template.md, we can can just add that it is deprecated and removed.

**Review by vpnachev — DISMISSED** [review]
/lgtm

**Review by ScheererJ — DISMISSED** [review]
/lgtm

**Review by ialidzhikov — DISMISSED** [review]
/lgtm

**Review by vpnachev — DISMISSED** [review]
/lgtm

**Review by vpnachev — APPROVED** [review]
/lgtm

**Pull request merged**